Description: fix regression for CMYK profiles
Author: Marti Maria, info@littlecms.com
Origin: upstream, https://github.com/mm2/Little-CMS/commit/328c226af654adb321cedaa5cbe470eebd10e021
Bug: https://github.com/mm2/Little-CMS/issues/345
Forwarded: not-needed
Last-Update: 2023-01-04
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/plugins/fast_float/src/fast_8_tethra.c b/plugins/fast_float/src/fast_8_tethra.c
index 84c36a0..a843813 100644
--- a/plugins/fast_float/src/fast_8_tethra.c
+++ b/plugins/fast_float/src/fast_8_tethra.c
@@ -364,9 +364,8 @@ cmsBool Optimize8BitRGBTransform(_cmsTransform2Fn* TransformFn,
     // Only on RGB
     if (T_COLORSPACE(*InputFormat)  != PT_RGB) return FALSE;
    
-    // This optimization only works on RGB8->RGB8 or RGB8->CMYK8
-    if (T_COLORSPACE(*OutputFormat) != PT_RGB &&
-        T_COLORSPACE(*OutputFormat) != PT_CMYK) return FALSE;
+    // This optimization only works on RGB8->RGB8
+    if (T_COLORSPACE(*OutputFormat) != PT_RGB) return FALSE;
 
     OriginalLut = *Lut;
    
@@ -428,7 +427,7 @@ cmsBool Optimize8BitRGBTransform(_cmsTransform2Fn* TransformFn,
         if (TransReverse[t] == NULL) goto Error;
     }
 
-    // Now inset the reversed curves at the begin of transform
+    // Now insert the reversed curves at the begin of transform
     LutPlusCurves = cmsPipelineDup(OriginalLut);
     if (LutPlusCurves == NULL) goto Error;
 
